src/amm/uniswap_v3/batch_request/mod.rs

```diff
@@ -197,40 +198,49 @@ where
     N: Network,
     P: Provider<T, N>,
 {
-    let deployer = ISyncUniswapV3PoolBatchRequest::deploy_builder(provider, pool.address);
+    let deployer = ISyncUniswapV3PoolBatchRequest::deploy_builder(provider, vec![pool.address]);
     let res = deployer.call_raw().await?;
 
-    let constructor_return = DynSolType::Tuple(vec![
+    let constructor_return = DynSolType::Array(Box::new(DynSolType::Tuple(vec![
         DynSolType::Uint(128),
         DynSolType::Uint(160),
         DynSolType::Int(24),
         DynSolType::Int(128),
-    ]);
+    ])));
+
     let return_data_tokens = constructor_return.abi_decode_sequence(&res)?;
 
-    if let Some(tokens_tup) = return_data_tokens.as_tuple() {
-        if tokens_tup[1]
-            .as_uint()
-            .ok_or(AMMError::BatchRequestError(pool.address))?
-            .0
-            .is_zero()
-        {
-            return Err(AMMError::BatchRequestError(pool.address));
+    if let Some(tokens_arr) = return_data_tokens.as_array() {
```

Comment:
Isn't this the exact same as the previous impl? Why is it necessary to wrap the Tuple in and Array explicitly? `tokens_arr.len()` will always equal 1 since this is hardcoded

Reply:
we still need to decode as array at first even if size to be 1. Bc the way how solidity encodes array, len, offset, data...

Reply:
Before we had some other contract, probably with just an Address as a constructor arg, hence, we had it without array for output. But when i recompiled contracts, i found out that old abi was not matching the contract.

Reply:
Ohhh got it, the ABI changed 👌